CONVENTIONS.md

```diff
@@ -0,0 +1,124 @@
+# Naming conventions
+
+## General
+
+Use `UpperCamelCase` for class names. Use `lowerCamelCase` for method names, variable names, and names of fields that are not
+both static and final. Always use `UPPER_SNAKE_CASE` for names of fields that are both static and final.
+
+Method names should always be verbal phrases (`tick`, `getCarversForStep`), except for builder methods and "withX" methods.
+Class names and non-boolean field and variable names should be noun phrases (`ChunkRegion`, `color`). Boolean field and
+variable names should always be adjectival phrases or present tense verbal phrases (`powered`, `canOpen`), avoiding the `is`
+and `has` prefixes when possible (`colored`, not `isColored` or `hasColor`).
+
+To make code as easy to read as possible, keep names in the natural language order. For example, a class representing a "chest
+block entity" should be named `ChestBlockEntity`, not `BlockEntityChest`. Though prefix naming may be helpful for grouping
+classes together in an IDE's tree view, reading and writing code is done much more often than browsing files.
+
+Use American English (`color`, not `colour`) for consistency throughout Yarn and with known Mojang names.
+
+Omit words that are unnecessary thanks to parameter names or owner class names. For example, use `getChunk(BlockPos pos)`
+rather than `getChunkAtPosition(BlockPos pos)`, and `create` rather than `createBox` for a method is in the `Box` class.
+Don't avoid overloading methods or shadowing fields.
+
+Names should be as descriptive as possible, so don't omit important words. When naming something always look at all its usages
+(including overriding methods and inheriting classes).
+
+Consistency is important as it makes code more readable and names easier to memorize. When possible, use terms that are present
+in other Yarn names, in libraries used by Minecraft, or in vanilla strings. See the "Common names" and "Mojang names" sections
+for more information.
+
+## Abbreviations
+
+Avoid abbreviations unless it's a common one everyone knows and other yarn names involving the same word use its abbreviated
+form. Full names are easier to read quickly and remember ("Which words were abbreviated?") and they often don't take more
+time to type thanks to IDE autocompletion. Common abbreviations you should use are:
+
+ - "id" for "identifier"
+ - "pos" for "position"
+ - "NBT" for "named binary tag"
```

Comment:
`"nbt"`

Reply:
I'm just giving the English acronym we use for "named binary tag". In code, it should be `nbt`, as described below ("Treat acronyms as single words rather than capitalising every letter.")

Reply:
I think it's a little confusing to have it capitalized here tbh. Should just show lowercase there too

Reply:
Done